Makefile

```diff
@@ -1,5 +1,13 @@
 ENABLE_FEATURES ?= default
 
+ifeq ($(ROCKSDB_SYS_STATIC),1)
```

Comment:
Now we have a `static_release` tag, so I think it is better to handle these in `static_release` directly.

For outside, `make release` is to build dynamic binary, and `make static_release` is to build static binary, no need to pass environment explicitly.